### DISCUSSION TITLE
[globalTypes] I'm looking for some way for using "control : color" for globaltype.

### DISCUSSION THREAD

**Discussion by rexgenPlatform** [storybookjs/storybook#28644] [Help]
### Summary

I'm trying to make some theme color palette for our designers.
So I tried to make global type in toolbar and read those in each stories.
But, global types aren't working proper in option of `control : color`.
It works for option of items, but not working in controls... so I'm asking in this part.

below is my preview.ts
(Storybook 8.1.10, React 18.3.1)
```
import type { Preview } from "@storybook/react";

const preview: Preview = {
  parameters: {
    controls: {
      matchers: {
        color: /(mainColor|subColor|color|backgroundColor|hoverColor|hoverBackgroundColor)$/i,
        date: /Date$/i,
      },
    },
  },
  globalTypes: {
    mainColor: {
      description: '전체 테마의 색상 변경',
      defaultValue: '#ff0000',
      toolbar: {
        title: 'Primary',
        items: [ '#ff0000', '#00ff00', '#0000ff']
      }
    },
    subColor: {
      description: '서브 테마의 색상 변경',
      defaultValue: '#000000',
      toolbar: {
        title: 'Sub',
      }
    },
  },
};

export default preview;
```

and here is error

> Uncaught TypeError: Cannot read properties of undefined (reading 'map')
    at J (manager-bundle.js:2:4580)
    at manager-bundle.js:2:4877
    at Array.map (<anonymous>)
    at Q (manager-bundle.js:2:4863)
    at renderWithHooks (chunk-B3YDJJJH.js:92:907)
    at updateFunctionComponent (chunk-B3YDJJJH.js:96:14955)
    at beginWork (chunk-B3YDJJJH.js:98:28838)
    at HTMLUnknownElement.callCallback2 (chunk-B3YDJJJH.js:31:42734)
    at Object.invokeGuardedCallbackImpl (chunk-B3YDJJJH.js:31:43261)
    at invokeGuardedCallback (chunk-B3YDJJJH.js:31:44569)

**Comment by valentinpalkovic** [comment]
@ndelangen mentioning you here because you are currently working on globals. Maybe you have some insights.

**Comment by ndelangen** [comment]
Please show to code that's actually failing, there's something on the manager's side where you're calling `.map`.

Most likely you need to ensure that code is ran optionally, because globals start out undefined in the manager.
